contracts/tests/test_locker.cairo

```diff
@@ -0,0 +1,119 @@
+use unruggablememecoin::locker::{ILockerDispatcher, ILockerDispatcherTrait};
+use unruggablememecoin::sample_erc20::{SampleERC20, IERC20Dispatcher, IERC20DispatcherTrait};
+use result::ResultTrait;
+use snforge_std::{ declare, ContractClassTrait, start_prank, stop_prank, PrintTrait, start_warp, stop_warp, CheatTarget};
+use array::ArrayTrait;
+use starknet::ContractAddress;
+
+struct Setup {
+    locker: ILockerDispatcher,
+    lp_token: IERC20Dispatcher,
+    owner: ContractAddress
+}
+
+fn setup() -> Setup {
+    let owner_address: felt252 =
+        0x123123123123123123123123123;
+    let locker_contract = declare('Locker');
+
+    let mut locker_calldata = ArrayTrait::new();
+    locker_calldata.append(200_u64.into()); // 200ms lock
+
+    let locker_address = locker_contract.deploy(@locker_calldata).unwrap();
+
+    let lp_contract = declare('SampleERC20');
+
+    let mut lp_calldata = ArrayTrait::new();
+    lp_calldata.append('TEST');
+    lp_calldata.append('TST');
+    lp_calldata.append(18.into());
+    lp_calldata.append(owner_address); // Owner
+    lp_calldata.append(5000.into()); // 5000 tokens
+    lp_calldata.append(0.into());
+
+    let lp_address = lp_contract.deploy(@lp_calldata).unwrap();
+
+    // start_prank(locker_address, owner_address.try_into().unwrap());
+
+    start_prank(CheatTarget::One(lp_address), owner_address.try_into().unwrap());
+    IERC20Dispatcher { contract_address: lp_address }.approve(locker_address, u256 { low: 0xffffffffffffffffffffffffffffffff_u128, high: 0xffffffffffffffffffffffffffffffff_u128});
+    stop_prank(CheatTarget::One(lp_address));
+    Setup {
+        locker: ILockerDispatcher { contract_address: locker_address },
+        lp_token: IERC20Dispatcher { contract_address: lp_address },
+        owner: owner_address.try_into().unwrap()
+    }
+}
+
+#[test]
+fn test_lock_lp() {
+    let setup = setup();
```

Comment:
comment properly all tests and separate into clear steps for each test with comments like:
- Setup
- Mock
- Test
- Assertion